src/components/notebookRenderStrategy.tsx

```diff
@@ -121,4 +145,23 @@ export class NotebookRenderStrategy implements ExpandableNodeRenderStrategy {
 	private onChevronClick() {
 		this.expandNode();
 	}
+
+	private onExpand() {
+		if (!this.notebook.sections && !this.notebook.sectionGroups && this.notebook.apiUrl && this.globals.oneNoteDataProvider) {
+			this.globals.oneNoteDataProvider.getNotebookBySelfUrl(this.notebook.apiUrl, 5).then((notebook) => {
+				this.notebook.sections = notebook.sections
+				this.notebook.sectionGroups = notebook.sectionGroups
+			}).catch((apiError: any) => {
+				try {
+					this.notebook.apiHttpErrorMessage = JSON.parse(apiError.response).error.message
+				} catch (error) {
+					this.notebook.apiHttpErrorMessage = Strings.getError(apiError.statusCode);
+				}
+			}).then(() => {
+				if (this.globals.callbacks.onNotebookInfoReturned) {
```

Comment:
I still don't understand why this is here... why not put this after line 153? Actually, should this even happen if there was an error returned? Right now it does.

Reply:
discussed offline, this makes sense